### PR TITLE
PAL average volume sorting 

### PR DESCRIPTION
**Pull request by davidkelly**
Allow PAL to sort on a new average volume, which is the peak volume, with an exponential decay.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 51246a2e45ece2e60e11b97f388101de55a38800 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-51246a2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-51246a2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d938c70b486c26e9bff3c7dd78a1c9a86586098a available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-d938c70.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-d938c70.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 76229905f2eb6d5522af28d430e40dbdb06f3a84 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-7622990.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-7622990.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4a4c914ca8df6b1e7d7d66885789deac94263d31 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-4a4c914.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-4a4c914.dmg

**Comment by davidkelly** [comment]
please test this

**Comment by davidkelly** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4a4c914ca8df6b1e7d7d66885789deac94263d31 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-4a4c914.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-4a4c914.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2458f95c42398d904fa7d6e7e0829d864169bca0 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-2458f95.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-2458f95.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: baa7f81d793b59f771990be01545a7f3bda58047 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-baa7f81.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-baa7f81.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e7c8085bc8ff5c796a73502289a8e4d7473a4817 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-e7c8085.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-e7c8085.dmg

**Comment by davidkelly** [comment]
Test Plan

Bring up the PAL in a place where there are people talking.

(1) The VOL column has buttons with the average volume as the glyph.  It should vary with the speakers volume, as the peak with a decay.  You should be able to sort on this column, and find the loud/quiet people
(2) When you push the button, you are toggling the mute (what we called 'personal mute').  The glyph changes accordingly, to show you've muted them, but continues to display their average volume (like their vu meter)
(3) The VOL will vary with the setting of the gain slider too.  If you adjust the gain up/down, the VOL should vary accordingly.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3bf3f16a6ab53576d2d11d6dae901a8d0918e144 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-3bf3f16.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-3bf3f16.dmg

**Comment by howard-stearns** [comment]
In testing, don't get confused by https://highfidelity.fogbugz.com/f/cases/3313/PAL-audio-level-display-pops-back-to-gain-of-0-whenever-you-open-PAL-but-actual-gain-does-not, which is separate.

**Comment by howard-stearns** [comment]
In testing, don't get confused by https://highfidelity.fogbugz.com/f/cases/3314/sorting-by-table-column-in-qml-causes-buttons-to-be-highlighted-e-g-in-pal, which is separate.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3bf3f16a6ab53576d2d11d6dae901a8d0918e144 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-3bf3f16.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-3bf3f16.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c04fd0ec66d31d6a0476be57d81cdc8fbc59e358 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-c04fd0e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-c04fd0e.dmg

File: scripts/system/pal.js [hunk 1]
```diff
@@ -255,6 +266,8 @@ function fromQml(message) { // messages are {method, params}, like json-rpc. See
             UserActivityLogger.palAction("avatar_gain_changed", data['sessionId']);
         } else {
             Users.setAvatarGain(data['sessionId'], data['gain']);
+            sessionGains[data['sessionId']] = convertDbToLinear(data['gain']);
+            print("set " + data['sessionId'] + " to " + sessionGains[data['sessionId']]);
```
Comment (howard-stearns) [inline scripts/system/pal.js]: for next time:  foo['someLiteralString'] is rejected by linters, which prefer foo.someLiteralString.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1fc57ce9a68a40de030890c7e2aefb73b3de5375 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-1fc57ce.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-1fc57ce.dmg

**Comment by davidkelly** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1fc57ce9a68a40de030890c7e2aefb73b3de5375 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-1fc57ce.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-1fc57ce.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 55d049ff5caba783d1cbd9be59da2c53e52b518a available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-55d049f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-55d049f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 45293ab130b0ea9366915c32fd0ecb1edc418ef8 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-45293ab.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-45293ab.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dc754cff5f1ce8422c31aedfccba0c74769535c0 available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-dc754cf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-dc754cf.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5be952c4e8e5479377bc7db783b88a70314da23c available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.dmg

**Comment by howard-stearns** [comment]
test this please

**Comment by howard-stearns** [comment]
(turns out that last 'test this please' was unnecessary. It just wasn't showing me the new build it had made)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5be952c4e8e5479377bc7db783b88a70314da23c available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5be952c4e8e5479377bc7db783b88a70314da23c available at: https://gist.github.com/7b3e61297b81011e7476a4c15c536a60<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9698-5be952c.dmg

**Pull request merged**